### PR TITLE
blackbox logging for biohazards

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Blackbox now gets:
If terrors, blob or xenomorphs survive to round end (or just before round end via nuke), it is now recorded. Also records if they are wiped out.
As such, if 6 terrors exist and station gets nuked, we presume terrors did it, and declare it a nuclear terror victory.
This is also done for shuttle docking round end, and admin forced round ends.
Terrors are recorded as the type they spawned, IE white terrors, queen, prince...

## Why It's Good For The Game

Better logging of how often xenomorphs / blob / specific terror varieties win / die.

Todo:

- [x] Test this with the database I am trying to set up as we speak.

## Testing

Compiling, trying to now database test.

## Changelog
:cl:
add: Biohazards now log wins / deaths to blackbox
/:cl:

### CODE REVIEW

File: code/__DEFINES/misc_defines.dm [hunk 1]
```diff
@@ -577,3 +577,12 @@
...
 #define text_ref(datum) (isdatum(datum) ? (datum:cached_ref ||= "\ref[datum]") : ("\ref[datum]"))
+
+#define NUCLEAR 1
+#define CREW_TRANSFER 2
+#define FORCED_ROUND_END 3
+#define GREEN_SPIDER 1
+#define PRINCE_SPIDER 2
+#define WHITE_SPIDER 3
+#define PRINCESS_SPIDER 4
+#define QUEEN_SPIDER 5
```
Comment (Contrabang) [inline code/__DEFINES/misc_defines.dm:588]: These defines need to be way less generalized
```suggestion
#define ROUND_END_NUCLEAR 1
#define ROUND_END_CREW_TRANSFER 2
#define ROUND_END_FORCED 3

#define TS_INFESTATION_GREEN_SPIDER 1
#define TS_INFESTATION_PRINCE_SPIDER 2
#define TS_INFESTATION_WHITE_SPIDER 3
#define TS_INFESTATION_PRINCESS_SPIDER 4
#define TS_INFESTATION_QUEEN_SPIDER 5
```

**Review by Contrabang — CHANGES_REQUESTED** [review]

**Review by AffectedArc07 — APPROVED** [review]
Approved ticker stuff

File: code/controllers/subsystem/SSticker.dm [hunk 2]
```diff
@@ -768,3 +770,77 @@ SUBSYSTEM_DEF(ticker)
...
+				var/spiders = 0
+				for(var/mob/living/simple_animal/hostile/poison/terror_spider/S in GLOB.ts_spiderlist)
+					if(S.ckey)
+						spiders++
+				if(spiders >= 5 || (output == "Prince Terror" && spiders == 1)) //If a prince lives, record as win.
+					switch(outcome)
+						if(NUCLEAR)
+							SSblackbox.record_feedback("tally", "Biohazard nuclear victories", 1, output)
+						if(CREW_TRANSFER)
+							SSblackbox.record_feedback("tally", "Biohazard survives to nuclear round end", 1, output)
```
Comment (lewcc) [inline code/controllers/subsystem/SSticker.dm:800]: This isn't always a nuclear end though, right? Might be worth clarifying

Comment (Qwertytoforty) [reply code/controllers/subsystem/SSticker.dm:800]: Fixed

**Review by lewcc — APPROVED** [review]
Looks good

**Pull request merged**